onmt/tests/test_beam_search.py

```diff
@@ -502,11 +492,11 @@ def second_step(self, beam, expected_beam_scores, expected_len_pen):
     def third_step(self, beam, expected_beam_scores, expected_len_pen):
         # assumes beam 0 finished on last step
         scores_3 = torch.log_softmax(torch.tensor(
-            [[0, 0, 5000, 0, 5000, .51, .2, 0],  # beam 0 shouldn't cont
+            [[0, 0, 10000, 0, 5000, .51, .2, 0],  # beam 0 shouldn't cont
              [0, 0, 0, 0, 0, 0, 0, 0],
-             [0, 0, 0, 0, 0, 5000, 0, 0],
-             [0, 0, 0, .2, .2, .2, .2, .2],
-             [0, 0, 50, 0, .2, .2, .2, .2]]  # beam 4 -> beam 1 should die
+             [0, 0, 10000, 0, 0, 5000, 0, 0],
```

Comment:
Why is this change required ?
Same for the one next line

Reply:
The test was designed to have exactly n_best = 3 beams to finish, we need more beams to finish to satisfy the exit condition.
Before we had:
1 beam finishing in first step
1 beam finishing in second step
1 beam finishing in third step
total = 3
Now we need 5 beams to finish so I am changing the third step so that 3 beams finish in this step.